addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,122 @@
+package org.jboss.forge.addon.manager.impl.ui;
+
+import java.io.File;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.model.FileHeader;
+
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.Dependency;
+import org.jboss.forge.addon.dependencies.DependencyResolver;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.forge.addon.dependencies.builder.DependencyQueryBuilder;
+import org.jboss.forge.addon.resource.FileResource;
+import org.jboss.forge.addon.ui.command.AbstractUICommand;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.context.UIValidationContext;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.forge.furnace.versions.SingleVersion;
+import org.jboss.forge.furnace.versions.Version;
+
+public class DistributionUpdateCommand extends AbstractUICommand implements AddonCommandConstants
+{
+   @Inject
+   Addon currentAddon;
+   @Inject
+   private DependencyResolver dependencyResolver;
+
+   @Override
+   public UICommandMetadata getMetadata(UIContext ctx)
+   {
+      return Metadata.forCommand(getClass()).name("Forge Update Distribution")
+               .description("Update the whole forge installation")
+               .category(Categories.create(ADDON_MANAGER_CATEGORIES));
+   }
+
+   @Override
+   public Result execute(UIExecutionContext context) throws Exception
+   {
+      if (!context.getPrompt().promptBoolean(
```

Comment:
Also it would be nice to provide another command to abort the previous update. You could refactor this class to use the @Command annotation to add more commands instead of extending UICommand